src/components/LayerCollection/LayerPOLine.js

```diff
@@ -30,16 +32,56 @@ class LayerPOLine extends Component {
 
   constructor(props) {
     super(props);
+
+    this.state = {
+      openModal: false,
+    };
     this.transitionToParams = transitionToParams.bind(this);
     this.connectedPOLineForm = props.stripes.connect(POLineForm);
   }
 
-  submitPOLine(data) {
+  openModal = () => {
+    this.setState({
+      openModal: true,
+    });
+  }
+
+  closeModal = () => {
+    this.setState({
+      openModal: false,
+    });
+  }
+
+  submitPOLine = async (data) => {
     const newLine = cloneDeep(data);
     const { lineMutator, onCancel } = this.props;
 
-    lineMutator.POST(newLine).then(() => {
+    try {
+      const response = await lineMutator.POST(newLine);
+      if (!response.id) throw new Error(response);
       onCancel();
+    } catch (e) {
+      const errors = await e.json();
+      if (errors.errors.find(el => el.code === 'lines_limit')) this.openModal();
```

Comment:
please don't write like, it's unreadable. And here could be an error - `errors` response structure is not guaranteed.